integration/src/test/java/org/mule/test/core/context/notification/processors/ModuleComponentPathTestCase.java

```diff
@@ -621,6 +635,54 @@ public void flowWithScatterGather() throws Exception {
     assertNoNextProcessorNotification();
   }
 
+
+  @Description("Smart Connector inside a async scope")
+  @Issue("MULE-17020")
+  @Test
+  public void flowWithAsync() throws Exception {
+    flowRunner("flowWithAsyncAndSmartConnector").run();
+
+    DefaultComponentLocation flowComponentLocation = FLOW_WITH_ASYNC_AND_SC
+        .appendLocationPart("processors", empty(), empty(), OptionalInt.empty(), OptionalInt.empty())
+        .appendLocationPart("0", Optional.of(builder()
+            .identifier(ASYNC_IDENTIFIER)
+            .type(SCOPE).build()), CONFIG_FILE_NAME, of(113), of(9));
+
+
+    assertNextProcessorLocationIs(flowComponentLocation);
+
+    DefaultComponentLocation nestedSetHardcodedPayloadLocation = flowComponentLocation
+        .appendLocationPart("processors", empty(), empty(), OptionalInt.empty(), OptionalInt.empty())
+        .appendLocationPart("0", MODULE_SET_PAYLOAD_HARDCODED_VALUE, CONFIG_FILE_NAME, of(114), of(13));
+
+    DefaultComponentLocation setPayloadLocation = OPERATION_SET_PAYLOAD_HARDCODED_VALUE_FIRST_MP
+        .appendLocationPart("processors", empty(), empty(), OptionalInt.empty(), OptionalInt.empty())
+        .appendLocationPart("0", SET_PAYLOAD, MODULE_SIMPLE_FILE_NAME, of(13), of(13));
+
+    DefaultComponentLocation loggerComponentLocation = flowComponentLocation
+        .appendLocationPart("processors", empty(), empty(), OptionalInt.empty(), OptionalInt.empty())
+        .appendLocationPart("1", LOGGER, CONFIG_FILE_NAME, of(117), of(13));
+
+    List<String> locations = new LinkedList<>(asList(nestedSetHardcodedPayloadLocation.getLocation(),
+                                                     setPayloadLocation.getLocation(), loggerComponentLocation.getLocation()));
+
+    check(POLLING_TIMEOUT, POLLING_DELAY, () -> {
+      try {
+        ComponentLocation componentLocation = getMessageProcessorNotification().getComponent().getLocation();
+        if (locations.contains(componentLocation.getLocation())) {
+          locations.remove(locations.indexOf(componentLocation.getLocation()));
+        }
+        return locations.isEmpty();
+      } catch (AssertionError e) {
+        return false;
```

Comment:
better to propagate the assertion error, this way it is hidden on the failures report, eventually